src/core/annotation.js

```diff
@@ -2291,6 +2292,17 @@ class LineAnnotation extends MarkupAnnotation {
 
       const borderWidth = this.borderStyle.width;
 
+      // If the /Rect-entry is empty, create a fallback rectangle such that we
+      // get similar rendering/highlighting behaviour as in Adobe Reader.
+      if (isArrayEqual(this.rectangle, [0, 0, 0, 0])) {
+        this.rectangle = [
+          this.data.lineCoordinates[0] - 2 * borderWidth,
```

Comment:
While it's certainly an improvement that we now actually render the line, I can't help but notice that the canvas rendering is a quite off when compared to the annotation layer rendering. Do you have any idea what could cause that?

Reply:
Yes, it's a tiny bit off; however it's *not* perfect in Adobe Reader either and given that now we create the same /BBox I think that this is good enough all things considered (since it's the PDF document *itself* which is at fault here).

This is really the best I can come up with, but you're more than welcome to improve it :-)

Reply:
Ah, ok. I already expected that a bit, but I figured it couldn't hurt to ask since I can't check how Adobe Reader renders it. Thanks!

Reply:
Trust me, I really tried to make it fit *better* but I wasn't able to improve the AnnotationLayer-part without impacting the actual *rendering* negatively :-(
At some point, I just gave up and said to myself that it'll have to be "good enough" for now!

Reply:
It's definitely an improvement over rendering nothing, and given that the PDF document itself is corrupted I think this is more than fine for now :-)